tsl/test/expected/compression_ddl.out

```diff
@@ -980,3 +980,39 @@ WHERE hypertable_name = 'test1' and attname = 'bigintcol' ;
  public            | test1           | bigintcol |                      1 |                      |             | 
 (1 row)
 
+-- test compression default handling
+CREATE TABLE test_defaults(time timestamptz, device_id int);
+SELECT create_hypertable('test_defaults','time');
+NOTICE:  adding not-null constraint to column "time"
+      create_hypertable      
+-----------------------------
+ (12,public,test_defaults,t)
+(1 row)
+
+ALTER TABLE test_defaults SET (timescaledb.compress,timescaledb.compress_segmentby='device_id');
+-- create 2 chunks
+INSERT INTO test_defaults SELECT '2000-01-01', 1;
+INSERT INTO test_defaults SELECT '2001-01-01', 1;
+-- compress first chunk
+SELECT compress_chunk(show_chunks) AS compressed_chunk FROM show_chunks('test_defaults') ORDER BY show_chunks::text LIMIT 1;
+             compressed_chunk             
+------------------------------------------
+ _timescaledb_internal._hyper_12_89_chunk
+(1 row)
+
+SELECT * FROM test_defaults ORDER BY 1;
+             time             | device_id 
+------------------------------+-----------
+ Sat Jan 01 00:00:00 2000 PST |         1
+ Mon Jan 01 00:00:00 2001 PST |         1
+(2 rows)
+
+ALTER TABLE test_defaults ADD COLUMN c1 int;
+ALTER TABLE test_defaults ADD COLUMN c2 int NOT NULL DEFAULT 42;
+SELECT * FROM test_defaults ORDER BY 1;
+             time             | device_id | c1 | c2 
+------------------------------+-----------+----+----
+ Sat Jan 01 00:00:00 2000 PST |         1 |    | 42
+ Mon Jan 01 00:00:00 2001 PST |         1 |    | 42
+(2 rows)
+
```

Comment:
Need test with insert into compressed chunk a) with a value for the new column, b) without  a value for the new column.
Followed by a recompress_chunk to ensure that works as expected.

Reply:
done